quickwit-search/src/fetch_docs.rs

```diff
@@ -29,15 +29,16 @@ use tantivy::{IndexReader, ReloadPolicy};
 use tracing::error;
 
 use crate::leaf::open_index_with_cache;
+use crate::service::SearcherContext;
 use crate::GlobalDocAddress;
 
 /// Given a list of global doc address, fetches all the documents and
 /// returns them as a hashmap.
-#[allow(clippy::needless_lifetimes)]
 async fn fetch_docs_to_map(
     mut global_doc_addrs: Vec<GlobalDocAddress>,
     index_storage: Arc<dyn Storage>,
     splits: &[SplitIdAndFooterOffsets],
+    searcher_context: Arc<SearcherContext>,
```

Comment:
Just like `self`, I'd pass the context argument first in all those functions.

Reply:
sounds a good idea yes